engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -607,14 +608,15 @@ protected void serviceStart() throws Exception
     }
 
     try {
-      Configuration config = getConfig();
+      YarnConfiguration config = new YarnConfiguration(getConfig());
       if (SecurityUtils.isStramWebSecurityEnabled()) {
-        config = new Configuration(config);
         config.set("hadoop.http.filter.initializers", StramWSFilterInitializer.class.getCanonicalName());
       }
       WebApp webApp = WebApps.$for("stram", StramAppContext.class, appContext, "ws").with(config).start(new StramWebApp(this.dnmgr));
       LOG.info("Started web service at port: " + webApp.port());
-      this.appMasterTrackingUrl = NetUtils.getConnectAddress(webApp.getListenerAddress()).getHostName() + ":" + webApp.port();
+      String scheme = ConfigUtils.getSchemePrefix(config);
+      String hostname = NetUtils.getConnectAddress(webApp.getListenerAddress()).getHostName();
+      this.appMasterTrackingUrl = scheme + hostname + ":" + webApp.port();
```

Comment:
This is changing the contract of appMasterTrackingUrl in the web service response even though this is probably how it should have been in the first place. This would be a backwards incompatible change. Suggest to use a separate property for scheme.